### DISCUSSION TITLE
Explanation about hydration using layout with Apollo Client hook

### DISCUSSION THREAD

**Discussion by alimoli** [vercel/next.js#40229] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
 Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000
    Binaries:
      Node: 17.4.0
      npm: 8.19.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 12.2.6-canary.8
      eslint-config-next: N/A
      react: 18.1.0
      react-dom: 18.1.0
```

### Describe the Bug

The error message is quite common when developing with SSR.
However, I cannot explain me why this error comes in the following situation.

__app.tsx_
```
function App({Component, pageProps}: AppProps) {
    return (
        <AppProvider>
            <ApolloProvider client={client}>
                <Content>
                    <Header/>
                    <Component {...pageProps} />
                </Content>
                <Footer/>
            </ApolloProvider>
        </AppProvider>
    )
}
```

As you can see, I have the `Header` and `Footer` components that are always present and detached from specific pages:

- `Header` is a static component, no API calls, just styling.
- `Footer` is a CSR component as it calls the **API X** to populate the footer with some dynamic information.

All the pages use SSR without problems except one. The page having the error calls the same **API X** like the footer, but using SSR.

It seems like calling the same API at SSR and CSR causes some issues.
It is really strange because this SSR error comes **from the hydration of the `Footer` component** that should be not evaluated as SSR.

The page is rendered correctly anyway ✅

Therefore, why am I getting this error?

**UPDATE** 
In production, all the pages except the index/homepage have this problem in the console.
https://reactjs.org/docs/error-decoder.html/?invariant=418

I tried to follow this documentation (https://nextjs.org/docs/basic-features/layouts#data-fetching) by using `useEffect` with the Apollo client hook and the error is gone now.

Can I have an explanation?

### Expected Behavior

No error about hydration for a component that is in the layout and not in a SSR page.

### Link to reproduction

no-code

### To Reproduce

1. Have a layout where a `Footer` component calls API X using ApolloClient query hook
2. Have a SSR page that calls API X

**Comment by judicaelandria** [comment]
maybe I don't understand what you're trying to say here very well, but seems like it could be solved with dynamic import
